### PR TITLE
114 Connect to Spotify

### LINKED ISSUE #114: Connect to spotify
## Describe the feature
Allow spotify connection for users.

## TO DO
FRONT : link the spotify button to the spotify connexion page
BACK : add the user spotify token in DB

## Extra links (Optional)
_Add any other extra links_

## Screenshots (Optional)
_Add any screenshot_

### PR DESCRIPTION
**Pull request by 31Nathan**
# New feature
We can now log on to Spotify API with the client ID and secret.
New Actions : 
The user follows an Artist
The user is listening to music
The user is listening to specific music
New Reactions:
The music is paused
Playlist Creation
Shuffle mode

# Updated feature
# Bug fixed

### CODE REVIEW

**Review by JohanCDev — CHANGES_REQUESTED** [review]
Add the actions on the Areas.ts files of web and app

**Review by JohanCDev — DISMISSED** [review]
Lgtm

File: Server/Services/spotifyServices.js [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    'user-library-modify',
+    'user-library-read',
+    'user-top-read',
+    'user-read-playback-position',
+    'user-read-recently-played',
+    'user-follow-read',
+    'user-follow-modify'
+  ];
+
+const spotifyApi = new SpotifyWebApi({
```
Comment (STom6) [inline Server/Services/spotifyServices.js:46]: adds documentation

**Review by STom6 — COMMENTED** [review]
you have to add the description of your service in /Server/Services/description

**Review by CedricCORGE — CHANGES_REQUESTED** [review]
Faire la documentation dans le dossier Services pour ajouter Spotify au about.json

**Review by CedricCORGE — DISMISSED** [review]
NTR

File: Server/Services/spotifyServices.js [hunk 2]
```diff
@@ -0,0 +1,222 @@
+/**
+ * @constant SpotifyWebApi
+ * @module spotify-web-api-node
```
Comment (osvegn) [inline Server/Services/spotifyServices.js]: Module name have to be "spotifyService". Then you can make documentation for constant spotifyWebAPI

**Review by osvegn — CHANGES_REQUESTED** [review]
One bad documentation, otherwise it seems good

**Review by osvegn — APPROVED** [review]
Look good

**Review by CedricCORGE — APPROVED** [review]
NTR

**Pull request merged**